### ISSUE TITLE
Build fails when testing cross-compile

### ISSUE THREAD

**Issue by averissimo** [r-wasm/webr#233]
I'm having problems building with the custom LLVM, 

System: Ubuntu (Linux 6.2.0-23-generic)

```bash
sudo apt install emscripten cmake gperf liblzma-dev libpcre2-dev quilt  wget
git clone https://github.com/r-wasm/webr
cd webr
./configure
make
```

### Error

[config.log](https://github.com/r-wasm/webr/files/11846842/config.log)

```node
/home/averissimo/work/opensource/r-wasm/webr/libs/build/pcre2-10.39/build/conftest:147
      throw ex;
      ^

TypeError: Failed to parse URL from /home/averissimo/work/opensource/r-wasm/webr/libs/build/pcre2-10.39/build/conftest.wasm
    at Object.fetch (node:internal/deps/undici/undici:11753:11) {
  [cause]: TypeError [ERR_INVALID_URL]: Invalid URL
      at new NodeError (node:internal/errors:400:5)
      at URL.onParseError (node:internal/url:565:9)
      at new URL (node:internal/url:645:5)
      at new Request (node:internal/deps/undici/undici:7175:25)
      at fetch2 (node:internal/deps/undici/undici:10839:25)
      at Object.fetch (node:internal/deps/undici/undici:11751:18)
      at fetch (node:internal/process/pre_execution:237:25)
      at instantiateAsync (/home/averissimo/work/opensource/r-wasm/webr/libs/build/pcre2-10.39/build/conftest:1670:14)
      at createWasm (/home/averissimo/work/opensource/r-wasm/webr/libs/build/pcre2-10.39/build/conftest:1706:3)
      at Object.<anonymous> (/home/averissimo/work/opensource/r-wasm/webr/libs/build/pcre2-10.39/build/conftest:4987:11) {
    input: '/home/averissimo/work/opensource/r-wasm/webr/libs/build/pcre2-10.39/build/conftest.wasm',
    code: 'ERR_INVALID_URL'
  }
}

Node.js v18.13.0
```

**Comment by averissimo** [comment]
This was a problem by using Node.js v18 in emscripten (see https://github.com/emscripten-core/emscripten/issues/16915)

It's probably related to the version Ubuntu ships.

**Comment by georgestagg** [comment]
It looks like a newer version of Emscripten might also help -- I'm not sure which version Ubuntu ships in their repos.

In my setup, I use [nvm](https://github.com/nvm-sh/nvm) and [emsdk](https://github.com/emscripten-core/emsdk) to easily switch between versions of Node and Emscripten. Currently, I'm building with Node v20.1.0 and Emscripten v3.1.37, and I think our GHA scripts do the same.

Probably we should spend some time to investigate the exact minimum required version for both and update our `README.md` to reflect that.

**Comment by averissimo** [comment]
Yup! Node `v16` with `v3.1.37` _(or `v3.1.42`)_ also fails in building webR. It's the default version that comes with emscripten.

Node v20 solves all issues I was still facing! :tada: 

Thank you @georgestagg  :1st_place_medal: for detailing the versions you're using and pointing to the GHA

---

Additional help for others that may encounter problems building:

The error below (found in config.log) was solved by using node `v20` with emsdk `v3.1.37`

```node
TypeError: WebAssembly.Tag is not a constructor
    at Object.<anonymous> (/home/averissimo/work/opensource/r-wasm/webr/R/build/R-4.3.0/build/conftest:2:28937)
    at Module._compile (node:internal/modules/cjs/loader:1196:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1250:10)
    at Module.load (node:internal/modules/cjs/loader:1074:32)
    at Function.Module._load (node:internal/modules/cjs/loader:909:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:22:47
configure:27852: WARNING: cannot run mixed C/Fortran code
configure:27854: error: Maybe check LDFLAGS for paths to Fortran libraries?
```

**Comment by DivadNojnarg** [comment]
Hi,

Facing [issues](https://github.com/r-wasm/webr/issues/199#issuecomment-1616814315) with `node > v20` (tried multiple versions) and emcc `3.1.42` with similar error messages. 

Are there other versions prerequisites for all the necessary deps (below is what I have): 
- cmake (3.26.4)
- gperf (GNU gperf 3.0.3)
-  liblzma-dev (For Mac: xz (XZ Utils) 5.4.3; liblzma 5.4.3)
- quilt (0.67)
- wget (GNU Wget 1.21.1 built on darwin20.2.0.)
- libpcre2 (pcre2 from homebrew v10.42)
